### ISSUE TITLE
Underscore option does not apply to indexes

### ISSUE THREAD

**Issue by nbouvrette** [sequelize/sequelize#9737]
## What are you doing?

When trying to set a BTREE index on my model:
```js
const test = sequelize.define('testDatabase',
    {
        testField: Sequelize.STRING
    }, {
        underscored: true,
        indexes: [
            {
                name: 'testFieldIndex',
                method: 'BTREE',
                fields: ['testField']
            }
        ]
    }
);
```

## What do you expect to happen?
I would expect the DB table to be created with an index on the `test_field` (underscored) and possibly to have an index names `TEST_FIELD_INDEX` (standard seem to be capital for Mysql but `test_field_index` could work as well)

## What is actually happening?
`Unhandled rejection SequelizeDatabaseError: Key column 'testField' doesn't exist in table`

### This can be a workaround but it defeats a bit the idea of the underscore option.

```js
const test = sequelize.define('testDatabase',
    {
        testField: Sequelize.STRING
    }, {
        underscored: true,
        indexes: [
            {
                name: 'TEST_FIELD_INDEX',
                method: 'BTREE',
                fields: ['test_field']
            }
        ]
    }
);
```

__Dialect:__ Mysql (probably any)
__Dialect version:__ Mysql2 1.5.3
__Database version:__ 5.7.19
__Sequelize version:__ 5.0.0-beta.10
__Tested with latest release:__ Yes (beta)

**Note :** _Your issue may be ignored OR closed by maintainers if it's not tested against latest version OR does not follow issue template._

**Comment by sushantdhiman** [comment]
Have you tried creating index without passing any name?

**Comment by nbouvrette** [comment]
Yes, I guess this can be a decent alternative. Sorry for missing that the name was not mandatory - I will close the issue for now but it could still be an interesting feature to support when using the name field.

**Comment by nbouvrette** [comment]
Actually, this issue was so old... I completely forgot what it was about but I should have focused on the "fields" problem rather than "name" - I have updated the original comment with a better explanation of the issue.

**Comment by sushantdhiman** [comment]
You need to pass fields into `fields` option not `attribute`, so I say it is working properly

**Comment by nbouvrette** [comment]
I'm not sure we're talking about the same thing here, this issue was meant to reflect that when using the `underscore` option it doesn't work in the `fields` when setting indexes. Check the example, I would expect to be able to use `fields: ['testField']` but I need to use `fields: ['test_field']`.

**Comment by nbouvrette** [comment]
So from what you're saying, there is no way to support `underscore`  in indexes and no plan to ever do so? I'm not really familiar with the difference in how `attributes ` and `fields` but conceptually it would make sense to have all variables referring to column names support this feature to keep the code consistent.
